crates/pathfinder/src/state/sync.rs

```diff
@@ -821,7 +821,14 @@ async fn l2_update(
         let transaction_data = block
             .transactions
             .into_iter()
-            .zip(block.transaction_receipts.into_iter().map(Some))
+            .zip(block.transaction_receipts.into_iter())
+            .map(
+                |(tx, (receipt, events))| pathfinder_storage::TransactionData {
+                    transaction: tx,
+                    receipt: Some(receipt),
+                    events: Some(events),
+                },
```

Comment:
Do you think its worth "simplifying" the storage here to follow the p2p model? ie change the storage to take in only `Vec<Transaction>` and no optional receipt and events. First write all txns, then all receipts, then all events.

Might be measureably slower though 🤔 Or faster depending on how sqlite stores the columns on disk.

Reply:
Personally for me its more about reducing the API surface of storage if possible - less code, less bugs 🤞

@kkovaacs might have a better feeling for performance impact, if any?